File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution
{
    public int solution(int n, int a, int b)
    {
        int answer = 1;
        int temp = a + b;
        a = (int) Math.min(a, b);
        b = temp - a;
        
        while (a % 2 != 1 || a + 1 != b) {
            a = (a + 1) / 2;
            b = (b + 1) / 2;
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
def solution(n,a,b):
    answer = 1
    a, b = min(a, b), max(a, b)
    
    while a % 2 != 1 or b % 2 != 0 or a + 1 != b:
        a = (a + 1) // 2
        b = (b + 1) // 2
        answer += 1
        
    return answer
